f3/granite.go

```diff
@@ -460,6 +477,32 @@ func (i *instance) tryCommit(round uint32) {
 	}
 }
 
+// maybeAmplify implements Bracha amplification:
+// If a weak quorum of DECIDE messages has been received, at least one correct participant must have sent one
+// (and there exists a proof of it, albeit not necessarily locally present).
+// That means that this value must be terminated by every correct participant.
+// Broadcasting a DECIDE message is necessary for liveness if the broadcast primitive is only best-effort broadcast.
+// If broadcast() implemented reliable broadcast, this step would not be necessary.
+func (i *instance) maybeAmplify(value ECChain) {
```

Comment:
Nit: please move this method lower down so it does not interrupt the sequence of try/beginPhase methods.